### ISSUE TITLE
checkout broken for app repos using 'main' instead of 'master'

### ISSUE THREAD

**Issue by MichaelBrim** [olcf/olcf-test-harness#29]
I recently added the [applications/frontier/coral2-amg](https://gitlab.ccs.ornl.gov/olcf-system-test/applications/frontier/coral2-amg) repo. Its default branch is named 'main', while historically the app repos have used 'master'.

When trying to run `runtests.py -m checkout`, I get the following error during the repo clone:
```
===========================
Error in subprocess command: cd /gpfs/alpine/stf016/scratch/mjbrim/crusher/apps; git clone --branch master --recurse-submodules gitlab@gitlab.ccs.ornl.gov:olcf-system-test/applications/frontier/coral2-amg.git
===========================
Standard out:
b''

Standard error:
b"Cloning into 'coral2-amg'...\n
<snipped>
fatal: Remote branch master not found in upstream origin\n"

===========================
```

Unfortunately, this looks like a machine-level setting (i.e., the choice of 'master' is set in `configs/crusher.ini`). Other than performing a manual checkout, I don't see a workaround.

**Comment by verolero86** [comment]
- Application level config could be the place to include these and other features that are for an application. Need new issue for that.
- For this particular issue the easiest solution would be to remove `--branch master` so that the default branch of a repo is checked out but we should come up with a way to provide this as an option
- Setting RGT_GIT_REPS_BRANCH=“main” is a workaround but doesn't solve the issue.

**Comment by verolero86** [comment]
Fixed in #30. Thanks go to @MichaelBrim ! 👏
